internal/storage/writer/azure/azure.go

```diff
@@ -226,25 +226,20 @@ func (m *MultiAccountWriter) getContainerURL() (*azblob.ContainerURL, error) {
 }
 
 func getServicePrincipalToken(monitor monitor.Monitor) (*adal.ServicePrincipalToken, error) {
-
-	spt, err := adal.NewServicePrincipalTokenFromManagedIdentity(azure.PublicCloud.ResourceIdentifiers.Storage, &adal.ManagedIdentityOptions{})
-
-	if err == nil {
-		monitor.Info("azure: acquired Manange Identity Credentials")
-		return spt, err
+	if settings, err := auth.GetSettingsFromEnvironment(); err == nil {
+		if cc, err := settings.GetClientCredentials(); err == nil {
+			monitor.Info("azure: acquired Manange Identity Credentials from ENV")
```

Comment:
I think it would be better to use "acquired credentials from Environment" because the one below it is not a managed identity, but rather a service principal. I think it was an oversight before

Reply:
Done, thanks, good observation